src/queryHistory/queryHistoryProvider.ts

```diff
@@ -0,0 +1,206 @@
+        this.openQueryHistoryEntry(queryHistoryQuickPickItem.node, queryHistoryQuickPickItem.action ===
+            QueryHistoryAction.RunQueryHistoryAction);
+    }
+
+    /**
+     * Starts the history capture by changing the setting
+     * and changes context for menu actions
+     */
+    public async startQueryHistoryCapture(): Promise<void> {
+        await this._vscodeWrapper.setConfiguration(Constants.extensionConfigSectionName,
+            Constants.configEnableQueryHistoryCapture, true);
+        await this._vscodeWrapper.setContext(Constants.isQueryHistoryEnabled, true);
+    }
+
+    /**
+     * Pauses the history capture by changing the setting
+     * and changes context for menu actions
+     */
+    public async pauseQueryHistoryCapture(): Promise<void> {
+        await this._vscodeWrapper.setConfiguration(Constants.extensionConfigSectionName,
+            Constants.configEnableQueryHistoryCapture, false);
+        await this._vscodeWrapper.setContext(Constants.isQueryHistoryEnabled, false);
+    }
+
+    /**
+     * Opens a query history listing in a new query window
+     */
+    public async openQueryHistoryEntry(node: QueryHistoryNode, isExecute: boolean = false): Promise<void> {
+        const editor = await this._untitledSqlDocumentService.newQuery(node.queryString);
+        let uri = editor.document.uri.toString(true);
+        let title = path.basename(editor.document.fileName);
+        const queryUriPromise = new Deferred<boolean>();
+        let credentials = this._connectionManager.getConnectionInfo(node.ownerUri).credentials;
+        await this._connectionManager.connect(uri, credentials, queryUriPromise);
+        await queryUriPromise;
+        this._statusView.languageFlavorChanged(uri, Constants.mssqlProviderName);
+        this._statusView.sqlCmdModeChanged(uri, false);
+        if (isExecute) {
+            const queryPromise = new Deferred<boolean>();
+            await this._outputContentProvider.runQuery(this._statusView, uri, undefined, title, queryPromise);
+            await queryPromise;
+            await this._connectionManager.connectionStore.removeRecentlyUsed(<IConnectionProfile>credentials);
+        }
+    }
+
+    /**
+     * Deletes a query history entry for a URI
+     */
+    public deleteQueryHistoryEntry(ownerUri: string): void {
+        let index = this._queryHistoryNodes.findIndex(node => {
+            return (node as QueryHistoryNode).ownerUri === ownerUri;
+        });
+        this._queryHistoryNodes.splice(index, 1);
+        this._onDidChangeTreeData.fire();
+    }
+
+    /**
+     * Getters
+     */
+    public get queryHistoryNodes(): vscode.TreeItem[] {
+        return this._queryHistoryNodes;
+    }
+
+    /**
+     * Limits the size of a string with ellipses in the middle
+     */
+    public static limitStringSize(input: string, forCommandPalette: boolean = false): string {
```

Comment:
This seems like something that can go in a generic utils file instead (if you do probably want to make the values params instead of being hardcoded like that. Explanations of why you chose these numbers would be useful too. What happens if you change the font size - do they still display correctly?)